### PR TITLE
Documentation using OpenAPI/Swagger for the API

### LINKED ISSUE #103: Create a documentation using OpenAPI/Swagger for the API
We need to describe the api endpoints on swagger, defining their request body, responses and more.

Link to doc:
https://swagger.io/docs/specification/basic-structure/

### PR DESCRIPTION
**Pull request by tonyaraujop**
This PR adds swagger documentation for API controllers.

image

Closes #103

### CODE REVIEW

File: app/controllers/api/v1/holidays_controller.rb [hunk 1]
```diff
@@ -1,8 +1,6 @@
 module Api
   module V1
-    class HolidaysController < ActionController::API
-      before_action :authenticate_user!
```
[the remark this answers is not in the document]
Comment (tonyaraujop) [reply app/controllers/api/v1/holidays_controller.rb:4]: All other controllers inheritance from ApiController and in order to swagger tests work properly this was necessary.

File: config/routes.rb [hunk 2]
```diff
@@ -41,16 +44,16 @@
...
-      get "users" => "companies#users"
-      get "offices" => "companies#offices"
-      get "holidays" => "holidays#holidays_dashboard"
-      post "request" => "token#request_token"
-      post "refresh" => "token#refresh_token"
+      get 'users' => 'companies#users'
+      get 'offices' => 'companies#offices'
+      get 'holidays' => 'holidays#holidays_dashboard'
+      post 'request' => 'token#request_token'
+      post 'refresh' => 'token#refresh_token'
```
[the remark this answers is not in the document]
Comment (tonyaraujop) [reply config/routes.rb:56]: Yeah you are right. This was done after using rubocop and I must missed during commit.

File: app/controllers/api/v1/holidays_controller.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (adbatista) [reply app/controllers/api/v1/holidays_controller.rb:4]: ok, please check if the /dashboard is working and locking Holliday days

File: config/routes.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (adbatista) [reply config/routes.rb:56]: ok, since you have this PR open and already did changes on rubocop.yml
disable the single quotes cop please

**Review by adbatista — APPROVED** [review]

**Pull request merged**